apps/backoffice/src/lib/be/institutions/business.ts

```diff
@@ -0,0 +1,57 @@
+import { Institution as BackofficeInstitution } from "@/generated/api/Institution";
+import {
+  InstitutionResource,
+  InstitutionResources
+} from "@/types/selfcare/InstitutionResource";
+import { EmailString } from "@pagopa/ts-commons/lib/strings";
+import * as O from "fp-ts/lib/Option";
+import { pipe } from "fp-ts/lib/function";
+import { getInstitutionById, getUserAuthorizedInstitutions } from "./selfcare";
+
+export const retrieveUserAuthorizedInstitutions = async (
+  selfCareUserId: string
+): Promise<InstitutionResources> => {
+  const apiResult = await getUserAuthorizedInstitutions(selfCareUserId);
+  return apiResult.map(toInstitution);
+};
+export const retrieveInstitution = async (
+  institutionId: string
+): Promise<BackofficeInstitution> => {
+  return await getInstitutionById(institutionId);
+};
+
+const toInstitution = (
+  institutionResource: InstitutionResource
+): BackofficeInstitution => ({
+  id: institutionResource.id,
+  externalId: institutionResource.externalId,
+  originId: institutionResource.originId,
+  description: institutionResource.description,
+  mailAddress: pipe(
+    institutionResource.digitalAddress,
+    EmailString.decode,
+    O.fromEither,
+    O.toUndefined
+  ),
+  address: institutionResource.address,
+  zipCode: institutionResource.zipCode,
+  taxCode: institutionResource.taxCode,
+  origin: institutionResource.origin,
+  institutionType: institutionResource.institutionType,
+  //attributes: institutionResource.attributes,
+  paymentServiceProvider: institutionResource.pspData,
+  dataProtectionOfficer: institutionResource.dpoData,
+  //geographicTaxonomies: institutionResource.geographicTaxonomies,
+  rea: institutionResource.companyInformations?.rea,
+  shareCapital: institutionResource.companyInformations?.shareCapital,
+  businessRegisterPlace:
+    institutionResource.companyInformations?.businessRegisterPlace,
+  supportEmail: institutionResource.assistanceContacts?.supportEmail,
+  supportPhone: institutionResource.assistanceContacts?.supportPhone,
+  //imported: institutionResource.imported,
+  //logo: institutionResource.logo,
```

Comment:
**question:** are these comments left here intentionally?

Reply:
Removed in cleanUp dbecc499cf52f87fed0bebcf542f935a88a96c5e